sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp

```diff
@@ -151,22 +151,25 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
   PathClient FileSystemClient::GetPathClient(const std::string& path) const
   {
     auto builder = m_dfsUri;
-    builder.AppendPath(path, true);
+    builder.AppendPath(path);
     return PathClient(builder, m_blobContainerClient.GetBlobClient(path), m_pipeline);
   }
 
   FileClient FileSystemClient::GetFileClient(const std::string& path) const
   {
 
     auto builder = m_dfsUri;
-    builder.AppendPath(path, true);
-    return FileClient(builder, m_blobContainerClient.GetBlobClient(path), m_pipeline);
+    builder.AppendPath(path);
+    auto blobClient = m_blobContainerClient.GetBlobClient(path);
+    auto blockBlobClient = blobClient.GetBlockBlobClient();
+    return FileClient(
+        std::move(builder), std::move(blobClient), std::move(blockBlobClient), m_pipeline);
   }
 
   DirectoryClient FileSystemClient::GetDirectoryClient(const std::string& path) const
   {
     auto builder = m_dfsUri;
-    builder.AppendPath(path, true);
+    builder.AppendPath(path);
```

Comment:
@Jinming-Hu, similar to https://github.com/Azure/azure-sdk-for-cpp/pull/655#discussion_r489134444, are you planning to add tests to cover these as well, as part of https://github.com/Azure/azure-sdk-for-cpp/issues/664?

Reply:
@katmsft is responsible for datalake service.

Reply:
@katmsft - add unit tests?